servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -103,10 +104,16 @@
     }
 
     void inject(final Publisher<Object> lbEventStream) {
+        if (this.lbEventStream != null) {
+            throw new ConcurrentModificationException();
+        }
         this.lbEventStream = lbEventStream;
     }
 
     void inject(final Completable sdStatus) {
+        if (this.sdStatus != null) {
+            throw new ConcurrentModificationException();
```

Comment:
Is this even possible? a `create` method is invoked by the client builder, meaning nobody else has a reference to the filter to inject something

Reply:
I will remove these, it was only for debugging. Regardless, the problem is that the shared DEFAULT is re-injected state in test environment and concurrent testing and this is causing the problem. The factory is holding state. This shouldn't be a problem for real use-cases but nevertheless let's discuss about it before we merge changes.